pkg/statistics/handle/handle.go

```diff
@@ -171,50 +171,47 @@ func NewHandle(
 	return handle, nil
 }
 
-// GetTableStats retrieves the statistics table from cache, and the cache will be updated by a goroutine.
-// TODO: remove GetTableStats later on.
-func (h *Handle) GetTableStats(tblInfo *model.TableInfo) *statistics.Table {
-	return h.GetPartitionStats(tblInfo, tblInfo.ID)
+// GetPhysicalTableStats retrieves the statistics for a physical table from cache or creates a pseudo statistics table.
+// physicalTableID can be a table ID or partition ID.
+func (h *Handle) GetPhysicalTableStats(physicalTableID int64, tblInfo *model.TableInfo) *statistics.Table {
+	tblStats, found := h.getStatsByPhysicalID(physicalTableID, tblInfo)
+	intest.Assert(tblStats != nil, "stats shoud not be nil")
+	intest.Assert(found, "stats shoud not be nil")
+	return tblStats
 }
 
-// GetTableStatsForAutoAnalyze is to get table stats but it will not return pseudo stats.
-func (h *Handle) GetTableStatsForAutoAnalyze(tblInfo *model.TableInfo) *statistics.Table {
-	return h.getPartitionStats(tblInfo, tblInfo.ID, false)
+// GetNonPseudoPhysicalTableStats retrieves the statistics for a physical table from cache, but it will not return pseudo.
+// physicalTableID can be a table ID or partition ID.
+// Note: this function may return nil if the table is not found in the cache.
+func (h *Handle) GetNonPseudoPhysicalTableStats(physicalTableID int64) (*statistics.Table, bool) {
+	return h.getStatsByPhysicalID(physicalTableID, nil)
 }
 
-// GetPartitionStats retrieves the partition stats from cache.
-// TODO: remove GetTableStats later on.
-func (h *Handle) GetPartitionStats(tblInfo *model.TableInfo, pid int64) *statistics.Table {
-	return h.getPartitionStats(tblInfo, pid, true)
-}
-
-// GetPartitionStatsForAutoAnalyze is to get partition stats but it will not return pseudo stats.
-func (h *Handle) GetPartitionStatsForAutoAnalyze(tblInfo *model.TableInfo, pid int64) *statistics.Table {
-	return h.getPartitionStats(tblInfo, pid, false)
-}
-
-func (h *Handle) getPartitionStats(tblInfo *model.TableInfo, pid int64, returnPseudo bool) *statistics.Table {
-	var tbl *statistics.Table
+func (h *Handle) getStatsByPhysicalID(physicalTableID int64, tblInfo *model.TableInfo) (*statistics.Table, bool) {
 	if h == nil {
-		tbl = statistics.PseudoTable(tblInfo, false, false)
-		tbl.PhysicalID = pid
-		return tbl
+		if tblInfo != nil {
+			tbl := statistics.PseudoTable(tblInfo, false, false)
+			tbl.PhysicalID = physicalTableID
+			return tbl, true
+		}
+		return nil, false
 	}
-	tbl, ok := h.Get(pid)
+
+	tbl, ok := h.Get(physicalTableID)
 	if !ok {
```

Comment:
I think `if ok {return tbl, true}` will make it cleaner by reducing a nested block.